lib/types.ts

```diff
@@ -194,3 +195,19 @@ export function callVariadicCallback<T>(
 }
 
 export type Callback<T> = CallbackNoError<T> | CallbackWithError<T>;
+
+export type InternalBreadcrumb = {
+  timestamp: number;
+  level: "debug" | "info" | "warning" | "error";
+  type: "manual" | "navigation" | "click-event" | "request" | "console";
+  category: string;
+  message: string;
+  customData?: CustomData;
+  className?: string;
+  methodName?: string;
+  lineNumber?: number;
+};
+
+export type Breadcrumb = Partial<
+  Pick<InternalBreadcrumb, "level" | "category" | "message" | "customData">
```

Comment:
I haven't seen yet where and how the two types are used, but now realising that `Breadcrumb` is a partial pick type of `IntnernalBreadcrumb` I wonder if the naming should be even flipped around so that the complete type is `Breadcrumb` and the partial-picked one gets a name that expresses that it's a light version of a `Breadcrumb`.

Will come back to this comment once I have a clearer picture, but don't want to lose this thought.

Reply:
Ok, I think I'd essentially I'd rejig the naming of these types.

To me, the full type is a `Breadcrumb` and the picked one is something "different" as indicated in my previous comment.

Reply:
This `Breadcrumb` object is what the end user will see, while `InternalBreadcrumb` is what we send after adding some data. 

Users never see the `InternalBreadcrumb` directly, but I agree that it can be renamed to something else.

I'd keep the `Breadcrumb` exported type name as it is.

Reply:
Looking at the Raygun4Flutter implementation, the public class is `RaygunBreadcrumbMessage`, so I can use that as well.

Reply:
I went with `BreadcrumbMessage` for the public type, and `Breadcrumb` for the internal payload sent to Raygun.